src/pages/swap/Swap.tsx

```diff
@@ -196,20 +196,86 @@ function Swap(): ReactElement {
       let error: string | null = null
       let amountToReceive = Zero
       const amountMediumSynth = Zero
+      let path: undefined | string[] = undefined
       if (amountToGive.gt(tokenBalances[formStateArg.from.symbol] || Zero)) {
         error = t("insufficientBalance")
       }
       if (amountToGive.isZero()) {
         amountToReceive = Zero
       } else if (
-        formStateArg.swapType === SWAP_TYPES.DIRECT &&
-        swapContract != null
+        routerContract != null
       ) {
-        amountToReceive = await swapContract.calculateSwap(
-          formStateArg.from.tokenIndex,
-          formStateArg.to.tokenIndex,
-          amountToGive,
-        )
+        try {
+          dispatch(setSwapRouterInfo({
+            ...swapRouterInfo,
+            bestSwapPath: null,
+            isGettingBestPath: true,
+          }))
+          const gasEstimate = await routerContract.estimateGas.findBestPathWithGas(
+            amountToGive,
+            tokenFrom.addresses[chainId],
+            tokenTo.addresses[chainId],
+            1,
+            BigNumber.from(225),
+          )
+
+          console.debug(`Gas Estimate: ${gasEstimate.toString()}`);
+
+          const optimalPath = await routerContract.findBestPathWithGas(
+            amountToGive,
+            tokenFrom.addresses[chainId],
+            tokenTo.addresses[chainId],
+            1,
+            BigNumber.from(225),
+            { gasLimit: (Number(gasEstimate) + 2000).toString() }
```

Comment:
@mohsenParvar we should change this to increase by a percentage instead of a fixed amount.